### PR TITLE
Update cadl-ranch to latest version

### PR DESCRIPTION
**Pull request by zzhxiaofeng**
fixes https://github.com/Azure/autorest.typescript/issues/2141

update version of `@azure-tools/cadl-ranch-specs` to `0.26.1`
update version of `@azure-tools/cadl-ranch-expect` to `0.9.0`
update version of `@azure-tools/cadl-ranch` to `0.10.0`

Updated list:
1.azure-core-basic
2.model-property-optional
3.model-property-value-types

### CODE REVIEW

File: packages/typespec-ts/test/integration/azurecore.spec.ts [hunk 1]
```diff
@@ -101,6 +101,36 @@ describe("Azure Core Rest Client", () => {
...
+      assert.strictEqual(result.status, "200");
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get second item as page item", async () => {
+    try {
+      const result = await client.path("/azure/core/basic/second-item").get();
+      assert.strictEqual(result.status, "200");
```
Comment (MaryGao) [inline packages/typespec-ts/test/integration/azurecore.spec.ts:128]: you need to assert the detail for result.

File: packages/typespec-ts/test/integration/azurecore.spec.ts [hunk 2]
```diff
@@ -101,6 +101,36 @@ describe("Azure Core Rest Client", () => {
...
+    } catch (err) {
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get first item as page item", async () => {
+    try {
+      const result = await client.path("/azure/core/basic/first-item").get();
+      assert.strictEqual(result.status, "200");
```
Comment (MaryGao) [inline packages/typespec-ts/test/integration/azurecore.spec.ts:119]: you need to assert the detail for result.

File: packages/typespec-ts/test/integration/azurecore.spec.ts [hunk 3]
```diff
@@ -101,6 +101,36 @@ describe("Azure Core Rest Client", () => {
     }
   });
 
+  it("should list with parameters", async () => {
+    try {
+      const result = await client.path("/azure/core/basic/parameters").get({
+        queryParameters: { another: "Second" },
+        body: { inputName: "Madge" }
+      });
+      assert.strictEqual(result.status, "200");
```
Comment (MaryGao) [inline packages/typespec-ts/test/integration/azurecore.spec.ts:110]: you need to assert the detail for result.

File: packages/typespec-ts/test/integration/modelPropertyOptional.spec.ts [hunk 4]
```diff
@@ -305,6 +305,349 @@ describe("ModelsPropertyOptional Rest Client", () => {
...
+      assert.fail(err as string);
+    }
+  });
+
+  it("should get default string literal", async () => {
+    try {
+      const result = await client
+        .path("/type/property/optional/string/literal/default")
+        .get();
+      assert.strictEqual(result.status, "200");
```
Comment (MaryGao) [inline packages/typespec-ts/test/integration/modelPropertyOptional.spec.ts:325]: assert result detail and this rule would apply to rest of cases.

File: packages/typespec-ts/test/integration/modelPropertyOptional.spec.ts [hunk 5]
```diff
@@ -305,6 +305,349 @@ describe("ModelsPropertyOptional Rest Client", () => {
...
+      assert.fail(err as string);
+    }
+  });
+
+  it("should put default union float literal", async () => {
+    try {
+      const result = await client
+        .path("/type/property/optional/union/float/literal/default")
+        .put({
+          body: {}
```
Comment (MaryGao) [inline packages/typespec-ts/test/integration/modelPropertyOptional.spec.ts:643]: why this is empty?

File: packages/typespec-ts/test/integration/modelPropertyOptional.spec.ts [hunk 6]
```diff
@@ -305,6 +305,349 @@ describe("ModelsPropertyOptional Rest Client", () => {
...
+      assert.fail(err as string);
+    }
+  });
+
+  it("should put default string literal", async () => {
+    try {
+      const result = await client
+        .path("/type/property/optional/string/literal/default")
+        .put({
+          body: {}
```
Comment (MaryGao) [inline packages/typespec-ts/test/integration/modelPropertyOptional.spec.ts:349]: why empty?

File: packages/typespec-ts/test/integration/modelPropertyOptional.spec.ts (hunk 5 above)
Comment (zzhxiaofeng) [reply packages/typespec-ts/test/integration/modelPropertyOptional.spec.ts:643]: The default value is set to empty like the other cases.

File: packages/typespec-ts/test/integration/modelPropertyOptional.spec.ts (hunk 6 above)
Comment (zzhxiaofeng) [reply packages/typespec-ts/test/integration/modelPropertyOptional.spec.ts:349]: The default value is set to empty like the other cases. For more details check here: https://github.com/Azure/cadl-ranch/blob/main/packages/cadl-ranch-specs/http/type/property/optionality/mockapi.ts#L22

File: packages/typespec-ts/test/integration/azurecore.spec.ts (hunk 3 above)
Comment (zzhxiaofeng) [reply packages/typespec-ts/test/integration/azurecore.spec.ts]: I have updated the test file as you said.

File: packages/typespec-ts/test/integration/modelPropertyOptional.spec.ts (hunk 4 above)
Comment (zzhxiaofeng) [reply packages/typespec-ts/test/integration/modelPropertyOptional.spec.ts:331]: I have updated the test file as you said.

File: packages/typespec-ts/test/integration/azurecore.spec.ts (hunk 2 above)
Comment (zzhxiaofeng) [reply packages/typespec-ts/test/integration/azurecore.spec.ts]: I have updated the test file as you said.

File: packages/typespec-ts/test/integration/azurecore.spec.ts [hunk 7]
```diff
@@ -32,11 +31,23 @@ describe("Azure Core Rest Client", () => {
         },
         contentType: "application/json"
       });
+      if (isUnexpected(result)) {
+        throw Error("Unexpected status code");
+      }
       assert.strictEqual(result.status, "200");
+      if (result.status === "200") {
```
Comment (MaryGao) [inline packages/typespec-ts/test/integration/azurecore.spec.ts:38]: why we need this check? is there any status code?

Comment (zzhxiaofeng) [reply packages/typespec-ts/test/integration/azurecore.spec.ts]: I have update the test file as you said

**Review by MaryGao — APPROVED** [review]
LGTM

**Comment by MaryGao** [comment]
I will merge this pr first @qiaozha feel free to add your comments if any!

**Pull request merged**